lnwire/message_test.go

```diff
@@ -452,8 +453,24 @@ func newMsgFundingLocked(t testing.TB, r io.Reader) *lnwire.FundingLocked {
 
 	pubKey := randPubKey(t)
 
-	msg := lnwire.NewFundingLocked(lnwire.ChannelID(c), pubKey)
-	msg.ExtraData = createExtraData(t, r)
+	// When testing the funding_locked msg type in the WriteMessage
+	// function we need to populate the alias here to test the encoding
+	// of the TLV stream.
+	aliasScid := lnwire.NewShortChanIDFromInt(rand.Uint64())
+	msg := &lnwire.FundingLocked{
+		ChanID:                 lnwire.ChannelID(c),
+		NextPerCommitmentPoint: pubKey,
+		AliasScid:              &aliasScid,
+		ExtraData:              make([]byte, 0),
+	}
+
+	// When testing the funding_locked msg type in the ReadMessage function
+	// we need to populate the ExtraData field here with a valid TLV stream
+	// and cannot use random data.
+	recordProducers := []tlv.RecordProducer{&aliasScid}
+	err = lnwire.EncodeMessageExtraData(&msg.ExtraData, recordProducers...)
+	require.NoError(t, err, "unable to encode channel alias into "+
+		"tlv stream")
```

Comment:
IIUC, we don't need to call `EncodeMessageExtraData` here, since it will also be done inside `Encode`.

Reply:
Thanks you are right, its now part because we write it before 👍

Reply:
Though I think also including it in the extradata is the correct way of doing it, because normally when calling `newMsgFundingLocked` I would expect a correctly filled message of this type without having to know that I need to `encode` it first so the TLV data is included in the right place ?

Reply:
but I am good with both ways